### ISSUE TITLE
When using latest GEOS model in the DAS must use O-server

### ISSUE THREAD

**Issue by bena-nasa** [GEOS-ESM/GEOSadas#88]
Another issue related to updating the GEOSadas to use newer MAPL/Model versions

With changes to History to use the new O-Server and ESMF regridding, at the GEOSadas scales (c720, ~80 collections) it is necessary to use the o-server to achieve optimal performance. We have been benchmarking the DAS history with the latest version of MAPL that has the newest o-server options.
It was found that on the skylakes, 8 server nodes in this configuration provided good performance with limited gains beyond 8 nodes:
mpirun  -np 5720 ./GEOSgcm.x --npes_model 5400 --nodes_output_server 8  --oserver_type   multigroup --npes_backend_pernode 8 --fast_oclient true

**Comment by rtodling** [comment]
Ben: What's the GCM layout in this configuration (NX, NY)?
Also: I assume you are referring to the version of the GCM in FP (5.27.1) ... which does not use MAPL2.0 - am I right?

**Comment by rtodling** [comment]
Question on this: how does the total number of PEs about relate to the ncpus-per-nodes?  So, we are on skylake w/ ncpus-per-node=40 but requesting a batch job w/ ncpus-per-node=36 would the math leading up to the number 5720 above change, that is,

5720 = 5400 + (8 * 40)

in the case of 36 should I have the entry in the option -np change to 

5400 + (36*8) = 5688

?

**Comment by bena-nasa** [comment]
Yes, if you are only running 36 cores per node then you would change the -np to 5400 + (36*8) = 5688
In general the -np XXX needs to be equal to or great than: (npes_model = nx*ny + (--nodes_output_server)*cores_per_node

**Comment by bena-nasa** [comment]
More clarification, the infrastructure is detecting the node information and partitioning the MPI communicators between the nodes as appropriate. The npes_model does not have to be evenly divisible by the number of nodes and if there are few left over mpi tasks, those just aren't used by either the model or o-server. So just to makeup numbers, lets say you had 23 cores per node and needed 540 mpi tasks for the model and wanted 4 O-server nodes, you would use
-np 644, --npes_model 540, --nodes_output_server 4. 

You need 24 nodes to run the model (so 12 mpi tasks just don't get used by in the actual GEOSgcm.x run, they just wait for the rest to finish) 
In other words on nodes 1-23, use all 23 mpi ranks to run the model, on node 24, 11 mpi ranks are used for the model, 12 are not used at all, just wait for everyone else, and the ranks on nodes 25-28 are used for the o-sever
644 > 540 + (23*4)

**Comment by rtodling** [comment]
I am closing this as this has been a part of GEOS ADAS for a few version now - certain in the latest v5.42.x
